tests/component-tests/textToSpeech/js/TextToSpeechTests.js

```diff
@@ -31,113 +31,171 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+    fluid.tests.textToSpeech.testStop = function (tts) {
+        var that = tts;
+        jqUnit.assert("The onStop event should have fired");
+        jqUnit.assertFalse("Should not be speaking", that.model.speaking);
+        jqUnit.assertFalse("Nothing should be pending", that.model.pending);
+        jqUnit.assertFalse("Shouldn't be paused", that.model.paused);
+        jqUnit.assertDeepEq("The queue should be empty", [], that.queue);
+        that.cancel();
+    };
 
-        jqUnit.assertTrue("The Text to Speech component should have initialized", that);
-        jqUnit.assertFalse("Nothing should be speaking", that.model.speaking);
+    fluid.tests.textToSpeech.testPause = function (tts) {
+        var that = tts;
+        jqUnit.assert("The pause event should have fired");
+        jqUnit.assertTrue("Should be speaking", that.model.speaking);
+        jqUnit.assertFalse("Nothing should be pending", that.model.pending);
+        jqUnit.assertTrue("Should be paused", that.model.paused);
+    };
+
+    fluid.tests.textToSpeech.testResume = function (tts) {
+        var that = tts;
+        jqUnit.assert("The resume event should have fired");
+        jqUnit.assertTrue("Should be speaking", that.model.speaking);
         jqUnit.assertFalse("Nothing should be pending", that.model.pending);
         jqUnit.assertFalse("Shouldn't be paused", that.model.paused);
-        jqUnit.start();
-    });
+    };
 
-    fluid.tests.textToSpeech.issueTest("Start and Stop Events", function () {
-        jqUnit.expect(10);
-        var that = fluid.tests.textToSpeech.startStop();
-        that.queueSpeech("Testing start and end events");
-    });
+    fluid.tests.textToSpeech.bypassTest = function () {
+        jqUnit.test("Tests were skipped - browser does not appear to support TTS", function () {
+            jqUnit.assert("TESTS SKIPPED - browser does not support SpeechSynthesis");
+        });
+    };
 
-    // Chrome doesn't properly support pause which causes this test to break.
-    // see: https://code.google.com/p/chromium/issues/detail?id=425553&q=SpeechSynthesis&colspec=ID%20Pri%20M%20Week%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified
-    if (!window.chrome) {
-        fluid.tests.textToSpeech.issueTest("Pause and Resume Events", function () {
-            jqUnit.expect(8);
-            var that = fluid.tests.textToSpeech.pauseResume();
-            that.queueSpeech("Testing pause and resume events");
+    var ttsStatus;
+
+    jqUnit.asyncTest("Confirming if TTS is available", function () {
+        jqUnit.expect(1);
+        var ttsSupport = fluid.textToSpeech.checkTTSSupport();
+        ttsSupport.then(function (){
+            ttsStatus = "TTS is available";
+            fluid.tests.textToSpeech.ttsTestEnvironment();
+        }, function () {
+            ttsStatus = "TTS is unavailable";          
+            fluid.tests.textToSpeech.bypassTest();
         });
-    }
 
+        var ttsTest = function () {
```

Comment:
Ok, I think we are mixing up here the requirement to have executed the asynchronous fluid.textToSpeech.checkTTSSupport before beginning the test and the asynchronous tests themselves. The "test scheduling a test" workflow is ok (since this happens a lot in practice with QUnit) but can be cleaned up.

All that we need is for the body of ttsTest to execute after the ttsSupport promise resolution, and the reject case is already fully self-sufficient, all we are adding to it is an extra assertion message. The most compact way of setting this up is probably to insert a ttsSupport.then(jqUnit.start, jqUnit.start) at line 185, and eliminate all the other material.